### DISCUSSION TITLE
Will you keep updating?

### DISCUSSION THREAD

**Discussion by bunnypig** [Mahmud0808/Iconify#394] [General]
I wanted to start off by saying I really appreciate your work and this is in no way meant to be taken as "where's the update?" or anything negative, just a question.

I read in one of the closed issues that someone said that you will not update Iconify further and it's dead. I just wanted to check if that statement was true or not?

Again, I'm very sorry to ask such a question because I know you do this in your spare time but I really enjoy Iconify, it's an excellent tool!

**Comment by Mahmud0808** [comment]
The project is currently on hiatus. There are several reasons for this.

- Google is frequently changing things in their source every 4 months (QRP1, QPR2, QPR3). I can't single handedly keep up with so many changes. It's like rewriting everything from scratch without breaking old android versions/updates.
- Google changed the style resources for QS tiles. Now whenever I try to change them, it just crashes SystemUI. Even when I copy the source code without making any changes and try to apply that, it still crashes. I have no idea why it's happening. Stuck in this for months and every probable solution I tried has failed. This demotivated me a lot.
- Google also did some sneaky things which now doesn't allow building overlays from framework. It has been fixed in https://github.com/iBotPeaches/Apktool/issues/3428 Althought it's not usable on android, it indicates that this issue can be fixed. Till now I was using https://github.com/JonForShort/android-tools but I don't think this repository will be updated either. I don't have enough knowledge about these to build by my own.

I hope that answers why the project is on hiatus. Iconify will continue if I find solutions for the above mentioned (except first one) issues.

**Reply by bunnypig** [reply]
Thank you for replying.
I understand everything and I fully respect your reasoning behind this. I can only imagine how much work is behind this.
I do hope you'll be motivated again in the future because you're great at what you do.
But again, thank you for all of your hard work you've put into this. It's greatly appreciated!

**Reply by tadekdudek** [reply]
I found this: https://github.com/lzhiyong/android-sdk-tools
It seems to be up to date. This will not solve all problems, but may be helpful.

**Reply by Mahmud0808** [reply]
Looks like I will have to build it myself. Thanks for the repo link. Looks promising.

**Reply by tadekdudek** [reply]
In releases there are prebuilt binaries

**Reply by Mahmud0808** [reply]
> In releases there are prebuilt binaries

Still having the same error on those latest binaries. I guess google haven't made those compatible for latest framework yet.

**Reply by HiFIi** [reply]
Progress at least

**Comment by HiFIi** [comment]
Sucks huh?

**Comment by srt54558** [comment]
> - Google is frequently changing things in their source every 4 months (QRP1, QPR2, QPR3). I can't single handedly keep up with so many changes. It's like rewriting everything from scratch without breaking old android versions/updates.

Maybe AI can help at some point if used correctly?

> - Google changed the style resources for QS tiles. Now whenever I try to change them, it just crashes SystemUI. Even when I copy the source code without making any changes and try to apply that, it still crashes. I have no idea why it's happening. Stuck in this for months and every probable solution I tried has failed. This demotivated me a lot.

Regarding this, I am not a android dev, and don't have any android coding experience, but I'm a big android enthusiast and front end developer. The issue is not the styles for sure. Some problems I think may be causing the crash:
-A new Signature verification feature in android
-New code outside verifying the styles 
-Dependencies relying on a hardcoded path to the Styles in communication with systemUI
-Missing resources in the installation folder of Iconify?
-Enabling Verbose Logging may help also. Maybe also reverse engineering... Removing one thing after another slowly till it crashes

I'd really like to help, but don't know where to start.

**Reply by Mahmud0808** [reply]
> -A new Signature verification feature in android

Don't think so. Only changing the qs tile resource crashes systemui. Everything else is fine.

> -New code outside verifying the styles

Highly unlikely.

> -Dependencies relying on a hardcoded path to the Styles in communication with systemUI

Don't think so.

> -Missing resources in the installation folder of Iconify?

I even tried adding some extra resources which I never needed to before. Don't ask me why lol, I just tried everything.

> -Enabling Verbose Logging may help also. Maybe also reverse engineering... Removing one thing after another slowly till it crashes

We don't need to reverse engineer. Android is open source. Even copy pasting directly from source code also crashed systemui. So I have literally no idea what they did that's causing this issue.

**Reply by srt54558** [reply]
> > -A new Signature verification feature in android
> 
> Don't think so. Only changing the qs tile resource crashes systemui. Everything else is fine.
> 
> > -New code outside verifying the styles
> 
> Highly unlikely.
> 
> > -Dependencies relying on a hardcoded path to the Styles in communication with systemUI
> 
> Don't think so.
> 
> > -Missing resources in the installation folder of Iconify?
> 
> I even tried adding some extra resources which I never needed to before. Don't ask me why lol, I just tried everything.
> 
> > -Enabling Verbose Logging may help also. Maybe also reverse engineering... Removing one thing after another slowly till it crashes
> 
> We don't need to reverse engineer. Android is open source. Even copy pasting directly from source code also crashed systemui. So I have literally no idea what they did that's causing this issue.

Reverse engineering makes sense in a manner of quickly finding what makes the issue. Going thru the code takes a lot of time

**Reply by HiFIi** [reply]
And the dev does it all for free!!

**Comment by bunnypig** [comment]
Just wanted to give my thanks to you. 
Tried the new update and it's working wonderfully!
Thank you for continuing and great work!! Very appreciated and I am very impressed you got it working. Will donate a couple of coffees tomorrow. Well done!
